GeoLib/AnalyticalGeometry-impl.h

```diff
@@ -0,0 +1,106 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+namespace GeoLib
+{
+
+template <class T_POINT>
+void getNewellPlane (const std::vector<T_POINT*>& pnts,
+                     MathLib::Vector3 &plane_normal,
+                     double& d)
+{
+    d = 0;
+    MathLib::Vector3 centroid;
+    std::size_t n_pnts(pnts.size());
+    for (size_t i(n_pnts - 1), j(0); j < n_pnts; i = j, j++) {
+        plane_normal[0] += ((*(pnts[i]))[1] - (*(pnts[j]))[1])
+                           * ((*(pnts[i]))[2] + (*(pnts[j]))[2]); // projection on yz
+        plane_normal[1] += ((*(pnts[i]))[2] - (*(pnts[j]))[2])
+                           * ((*(pnts[i]))[0] + (*(pnts[j]))[0]); // projection on xz
+        plane_normal[2] += ((*(pnts[i]))[0] - (*(pnts[j]))[0])
+                           * ((*(pnts[i]))[1] + (*(pnts[j]))[1]); // projection on xy
+
+        centroid += *(pnts[j]);
+    }
+
+    plane_normal.normalize();
+    d = MathLib::scalarProduct(centroid, plane_normal) / n_pnts;
+}
+
+template<class T_MATRIX>
+void compute2DRotationMatrixToX(MathLib::Vector3 const& v,
+        T_MATRIX & rot_mat)
+{
+    const double cos_theta = v[0];
+    const double sin_theta = v[1];
+    rot_mat(0,0) = rot_mat(1,1) = cos_theta;
+    rot_mat(0,1) = sin_theta;
+    rot_mat(1,0) = -sin_theta;
+}
+
+template <class T_MATRIX>
+void compute3DRotationMatrixToX(MathLib::Vector3  const& v,
+        T_MATRIX & rot_mat)
+{
+    // a vector on the plane
+    MathLib::Vector3 yy(0, 0, 0);
+    if (fabs(v[0]) > 0.0 && fabs(v[1]) + fabs(v[2]) < std::numeric_limits<double>::epsilon()) {
```

Comment:
It is also possible to use ```std::abs``` instead of ```fabs``` here and in line 55 as well as line 57 and 61.